src/libcore/iter/mod.rs

```diff
@@ -643,7 +643,8 @@ impl<A, B> FusedIterator for Chain<A, B>
 pub struct Zip<A, B> {
     a: A,
     b: B,
-    spec: <(A, B) as ZipImplData>::Data,
+    index: usize,
```

Comment:
could you add a comment here pointing out that these fields are only used for specialized instantiations of `A` and `B` (and point to #35727 in the comment for further explanation)